Modules/Cart/Http/Controllers/CartController.php

```diff
@@ -21,34 +22,94 @@ public function __construct(CartService $cartService)
     }
 
     /**
+     * Fetch logged user cart
+     *
      * @param Request $request
      * @return JsonResponse
      */
     public function fetch(Request $request): JsonResponse
     {
+
+        $user = auth()->user();
+        if ($user->cannot('view', Cart::class)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
         $response = $this->cartService->fetch($request);
+
         return response()->json($response);
     }
 
     /**
-     * @param Request $request
+     * User can add product on their cart
+     *
+     * @param CartRequest $request
      * @return JsonResponse
      */
-    public function add(Request $request): JsonResponse
+    public function add(CartRequest $request): JsonResponse
     {
-        $user = auth('sanctum')->user();
+        $user = auth()->user();
+        if ($user->cannot('create', Cart::class)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'Please login as a retailer',
+                'data' => ""
+            ]);
+        }
+
+        $response = $this->cartService->add($request);
 
-        $response = $this->cartService->add($request->all());
         return response()->json($response);
     }
 
     /**
+     * User can delete product from existing cart
+     *
      * @param Request $request
      * @return JsonResponse
      */
     public function delete(Request $request): JsonResponse
     {
+        $user = auth()->user();
+        $cart = Cart::where('id', $request->id)->first();
+        if ($user->cannot('delete', $cart)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'You are not authorized !',
+                'data' => ""
+            ]);
+        }
```

Comment:
for $response = $this->cartService->delete($request->id,$request->user_id); do not $request->user_id. From front end never send user_id.The code should be like this

$response = $this->cartService->delete($request->id)